ext/extconf.rb

```diff
@@ -1,28 +1,49 @@
 require 'mkmf'
 
-dir_config('R')
-
-some_paths = ENV['PATH'].split(File::PATH_SEPARATOR) + %w[
+# Searched after ENV['PATH']
+DEFAULT_R_PATHS = %w(
   /usr/local/lib64/R
   /usr/local/lib/R
   /usr/lib64/R
   /usr/lib/R
   /Library/Frameworks/R.framework/Resources
   /usr/local/Frameworks/R.framework/Resources
-]
+)
 
-some_lib_paths = some_paths.map{|dir| File.join(dir, 'lib') }
-find_library('R', nil, *some_lib_paths)
+# Searched after DEFAULT_R_PATHS/include
+DEFAULT_R_INCLUDE_PATHS = %w(
+  /usr/include/R
+  /usr/share/R/include
+)
 
-unless have_library("R")
+dir_config('R')
+
+r_paths = if $configure_args.has_key?('--with-R-dir')
+  [$configure_args['--with-R-dir']]
+else
+  r_paths = ENV['PATH'].split(File::PATH_SEPARATOR) + DEFAULT_R_PATHS
```

Comment:
Does this work? I'd expect `PATH` to include `.../bin` dirs, so we couldn't just add `lib` to these.

I think this might work better for the default case (no dir options specified):

```
append_cppflags(`R CMD config --cppflags`.shellsplit)
append_ldflags(`R CMD config --ldflags`.shellsplit)
```

Reply:
Yeah, that's a much better approach, I'll make that change. Just a heads up that it's a functionality change, if you look at what was there before.

Reply:
Made some changes, this builds and runs fine for me locally (assuming R_HOME is  _not_ set as an environment variable). If you don't specify flags at build time, we now just expect `R` to be there and for `R CMD` and `R RHOME` to work, and we use that to specify flags as you suggested. RSRUBY_R_HOME no longer defaults to NULL, we expect that `R RHOME` return something.